common/blobstore/interface.go

```diff
@@ -22,15 +22,19 @@ package blobstore
 
 import (
 	"context"
-	"errors"
+	"github.com/uber/cadence/.gen/go/shared"
 	"github.com/uber/cadence/common/blobstore/blob"
 )
 
 var (
-	// ErrBlobNotExists indicates that requested blob does not exist
-	ErrBlobNotExists = errors.New("requested blob does not exist")
 	// ErrBucketNotExists indicates that requested bucket does not exist
-	ErrBucketNotExists = errors.New("requested bucket does not exist")
+	ErrBucketNotExists = &shared.BadRequestError{Message: "requested bucket does not exist"}
+	// ErrBlobNotExists indicates that requested blob does not exist
+	ErrBlobNotExists = &shared.BadRequestError{Message: "requested blob does not exist"}
+	// ErrBlobSerialization indicates that a failure occurred in serializing blob
+	ErrBlobSerialization = &shared.BlobEncodingError{Message: "failed to serialize blob"}
```

Comment:
Do you need any special handling of BlobEncodingError?  If the intent is only to not treat it is Non-Retryable error then I would recommend to use BadRequestError here.